src/ngMessageFormat/messageformat.js

```diff
@@ -0,0 +1,938 @@
+'use strict';
+
+var angular = window['angular'];
+
+var $$messageFormatMinErr = angular['$$minErr']('$$messageFormat');
+
+// NOTE: ADVANCED_OPTIMIZATIONS mode.
+//
+// This file is compiled with Closure compiler's ADVANCED_OPTIMIZATIONS flag! Be wary of using
+// constructs incompatible with that mode.
+
+/**
+ * @ngdoc service
+ * @name $$messageFormat
+ *
+ * @description
+ * Angular internal service to recognize MessageFormat extensions in interpolation expressions.
+ * For more information, see:
+ * https://docs.google.com/a/google.com/document/d/1pbtW2yvtmFBikfRrJd8VAsabiFkKezmYZ_PbgdjQOVU/edit
+ */
+
+var noop = angular['noop'],
+    isFunction = angular['isFunction'],
+    toJson = angular['toJson'];
+
+function stringify(value) {
+  if (value == null) { return ''; }
+  switch (typeof value) {
+    case 'string':     return value;
+    case 'number':     return '' + value;
+    default:           return toJson(value);
+  }
+}
+
+var PARSE_CACHE_FOR_TEXT_LITERALS = new Object(null);
+var PARSE_CACHE_FOR_INTERPOLATIONS = new Object(null);
```

Comment:
Do you mean to use `Object.create(null)` here, which creates objects with no prototype, rather than `new Object(null)`, which is really just an empty object: `{}`?